### ISSUE TITLE
About the freeze_adapter()

### ISSUE THREAD

**Issue by dingfengqian** [SJTU-DMTai/OnlineTSF#4]
Hi, thanks for sharing this great codebase! While reading through the freeze_adapter() implementation, I noticed a potential issue regarding parameter freezing.
```
    def online(self, online_data=None, target_variate=None, phase='test', show_progress=False):
        if phase == 'val' and self.args.val_online_lr:
            lr = self.model_optim.param_groups[0]['lr']
            for j in range(len(self.model_optim.param_groups)):
                self.model_optim.param_groups[j]['lr'] = self.args.online_learning_rate
        self.model_optim.zero_grad()
        self._model.freeze_adapter(True)   # <-----
        ret = super().online(online_data, target_variate, phase, show_progress)
        self._model.freeze_adapter(False)   # <-----
        if phase == 'val' and self.args.val_online_lr:
            for j in range(len(self.model_optim.param_groups)):
                self.model_optim.param_groups[j]['lr'] = lr
        return ret
```

```
def freeze_adapter(self, freeze=True):
        for module_name in ['mlp1', 'mlp2']:
            if hasattr(self, module_name):
                getattr(self, module_name).requires_grad_(not freeze)
                getattr(self, module_name).zero_grad(set_to_none=True)
        for adapter in self.generator.bottlenecks.values():
            adapter.weights.requires_grad_(not freeze)
            adapter.weights.zero_grad(set_to_none=True)
            adapter.biases[:len(adapter.weights) - 1].requires_grad_(not freeze)
            adapter.biases[:len(adapter.weights) - 1].zero_grad(set_to_none=True)
```

In the current implementation, calling .requires_grad_(False) on modules like self.mlp1 and self.mlp2, which are instances of nn.Sequential, does not actually freeze the parameters inside. This is because nn.Sequential itself doesn't own parameters directly—its submodules do. As a result, calling self.mlp1.requires_grad_(False) will not affect the Linear layers inside, and those parameters will still receive gradients and be updated during training or online adaptation.

Additionally, the accompanying paper states: "During the online phase, we only fine-tune the forecast model and keep the parameters of the model adapter frozen." To align with this design, the freezing mechanism should ensure that no gradients are computed for the adapter parameters during online adaptation. Please let me know if I misunderstood something here—I'd really appreciate any clarification!

**Comment by dingfengqian** [comment]
I just realized this was due to a mistake on my side (misused requires_grad_() during debugging). The code is correct—I'm closing this issue now. Thanks!
